userprofile/templates/userprofile/profile.html

```diff
@@ -9,89 +9,114 @@
+		<div>
+			<label for="local-storage-services">{% trans "Automatically log in to:" %}</label>
+			<ul id="local-storage-services" class="local-storage-fields"
+				data-forget-text="{% trans "Forget automatic redirection" %}"></ul>
+		</div>
 		<div class="well">
-			<h4>{% trans "Automatic service redirections" %}</h4>
 			<p>
 				{% blocktrans %}
 				These are all services that you have marked for automatic redirection on this device.
-				You can remove automatic redirection here and you will be shown the privacy information next time you are being redirected to the service.
+				You can remove automatic redirection here and you will be shown the privacy information next time you
+				are being redirected to the service.
 				{% endblocktrans %}
 			</p>
-			<ul class="local-storage-fields" data-forget-text="{% trans "Forget automatic redirection" %}"></ul>
-			</dd>
 		</div>
-    </div>
-    <div class="col-md-4">
 
-        {# TODO: list enrolled courses #}
-    </div>
+		<form action="{% url 'set-user-language' %}" method="post" class="form form-horizontal">{% csrf_token %}
+			<div class="form-group">
+				<label class="col-xs-12" for="language-profile">{% trans 'Language' %}</label>
+				<div class="col-xs-12 col-sm-8">
+					<select name="language" class="form-control" id="language-profile">
+						{% get_current_language as LANGUAGE_CODE %}
+						{% get_available_languages as LANGUAGES %}
+						{% get_language_info_list for LANGUAGES as languages %}
+						{% for language in languages %}
+						<option value="{{ language.code }}" {% if language.code == LANGUAGE_CODE %}selected{% endif %}>
+							{{ language.name_local }}
+						</option>
+						{% endfor %}
+					</select>
+				</div>
+				<div class="col-xs-offset-0 col-xs-6 col-sm-4 ">
+					<button type="submit"
+						class="aplus-button--default aplus-button--md btn-block btn-save-profile">{% trans 'Save' %}</button>
+				</div>
+			</div>
+		</form>
+	</div>
 </div>
+<div class="row">
+	<div class="col-sm-offset-2 col-sm-8 col-md-4">
+		{% for item in backends.associated %}
+			{% if item.provider == 'google-oauth2' %}
+				<h2>{% trans "Information provided by Google:" %}</h2>
+			{% endif %}
+		{% endfor %}
+
+		{% for item in backends.not_associated %}
+			{% if item == 'google-oauth2' %}
+				<h2>{% trans "Information provided by Aalto:" %}</h2>
+			{% endif %}
+		{% endfor %}
+		<p class="text-warning">{% trans "The following settings should be changed with your identity provider" %}.</p>
+		<dl class="dl-horizontal">
+			{% trans "[Missing]" as miss %}
+			<dt class="provider-information">{% trans "Student ID" %}</dt>
+			<dd>{% firstof profile.student_id miss %}</dd>
+			<dt class="provider-information">{% trans "E-mail" %}</dt>
+			<dd>{% firstof user.email miss %}</dd>
+			<dt class="provider-information">{% trans "Username" %}</dt>
+			<dd>{% firstof user.username miss %}</dd>
+		</dl>
+	</div>
+	{# TODO: check the user type and show editing options/help #}
```

Comment:
Are these TODOs relevant anymore? The enrolled courses are now actually highlighted on the A+ front page.

Reply:
I started to use a local githook to stop pushes, when to code contains keyword "FIXME". For TODO, I have used the idea that it means "this could be made better, but it's not so important". So, from my point of view, every "FIXME" probably should have an issue. Some TODOs could have (i.e., things that should be cleaned after some time and such).